roles/ncn-common/files/scripts/metal/metal-lib.sh

```diff
@@ -196,29 +202,36 @@ function set_static_fallback() {
     local ipaddr
     local lan
     local netmask
-    local netconf=/tmp/netconf
+    local vendor
 
     # BMCs either run dedicated on lan3 (last LAN channel as is the case with Intel's),
     # or lan1 (when there's only one channel).
-    if ipmi_output_3=$(ipmitool lan print 3 2>/dev/null); then
-        lan=3
-        echo "$ipmi_output_3" > $netconf
-    elif [ -z $lan ] && ipmi_output_1=$(ipmitool lan print 1 2>/dev/null); then
-        lan=1
-        echo "$ipmi_output_1" > $netconf
-    elif [ -z $lan ]; then
-        echo "Failed to determine which LAN channel to use!"
+    printf 'scanning vendor ... '
+    vendor=$(ipmitool fru | grep -i 'board mfg' | tail -n 1 | cut -d ':' -f2 | tr -d ' ')
+    echo "$vendor"
+    case $vendor in
+        *'Intel'*'Corporation'*)
+            lan=3
+            ;;
+        *)
+            lan=1
+            ;;
+    esac
+    ipaddr=$(host $(hostname)-mgmt | awk '{print $NF}')
+    netmask=$(cidr_to_netmask $(craysys metadata get --level node ipam | jq -r '.hmn.ip' | awk -F / '{print $NF}'))
```

Comment:
If the cloud-init data is missing the hmn.ip field, the netmask will end up being 0.0.0.0.   We should handle this case to avoid setting a netmask of 0.0.0.0.

Reply:
We shouldn't get this far if that's the case, since `net-init.sh` would've failed.

I'll still handle the error for better defensive programming sake, and incase someone runs this ad-hoc.